src/stm32/stm32f0_serial.c

```diff
@@ -88,7 +88,9 @@
 
 #if CONFIG_MACH_STM32G0
   // The stm32g0 has slightly different register names
+  #ifdef LPUART2
   #define USART2_IRQn USART2_LPUART2_IRQn
+  #endif
   #define USART_CR1_RXNEIE USART_CR1_RXNEIE_RXFNEIE
```

Comment:
I don't understand what this does.  Is it that USART2_IRQn actually exists on the stm32g0b0?  If so, could this be written as:
```c
    #ifndef USART2_IRQn
      #define USART2_IRQn USART2_LPUART2_IRQn
   #endif
```

Reply:
The G0x0 family is missing some (or most) of the lower power peripherals. As such, these IRQ vectors that are combined might not be like that on the G0x0 family. In this case, the G0b0 doesn't have LPUART2, so the USART2 vector is just `USART2_IRQn`

Reply:
Unfortunately it can't be written as instructed since the IRQn thingies are actually part of an enum, not defines. So I had to revert to the old code. I did change the comment a bit however. Hopefully this makes the change a bit more clear.